### PR TITLE
Add MTE-1126 [v116] Add Slack notification for Robo test

### PR DESCRIPTION
**Pull request by AaronMT**
https://mozilla-hub.atlassian.net/jira/software/c/projects/MTE/boards/546

* Adds a Slack notification step
* Use secret for project set

Follow-up: I'd like to see how we can pull the web-link to a matrix exposed as part of the notification. From a notification, to getting a web-link will require many clicks.

Opening as a draft to check with @isabelrios to see if we only want notifications on failure (like Android), as well as which `fields:` we need.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -1419,14 +1419,34 @@ workflows:
...
-            $HOME/google-cloud-sdk/bin/gcloud config set project moz-firefox-ios-230118
+            $HOME/google-cloud-sdk/bin/gcloud config set project "$GOOGLE_CLOUD_PROJECT"
             $HOME/google-cloud-sdk/bin/gcloud beta firebase test ios run \
             --type robo \
             --app "$BITRISE_IPA_PATH" \
             --device=model=iphone13pro,version=15.7 \
             --results-bucket=firefox_ios_test_artifacts \
             --no-record-video \
             --client-details=matrixLabel="Bitrise"
+    - slack@3.1:
```
Comment (clarmso) [inline bitrise.yml:1430]: Have you tried a more recent version of the `slack` Bitrise step? The latest one is `3.5.0`. I may try to upgrade some Bitrise steps in the near future.

**Comment by isabelrios** [comment]
The PR looks good to me, I would try to add the link to the firebase results in it if we can get the url easily via script. Storing it in a ENV VAR and showing it in the slack notification is easy. So it may be doable without a lot of complexity...

File: bitrise.yml [hunk 2]
```diff
@@ -1419,14 +1419,34 @@ workflows:
...
+        - timestamp: 'no'
+        - fields: |
+            Task | ${BITRISE_BUILD_URL}
+            Commit | ${GIT_CLONE_COMMIT_MESSAGE_SUBJECT}
+            Source-Workflow | ${BITRISE_TRIGGERED_WORKFLOW_TITLE}
+            Branch | ${BITRISE_GIT_BRANCH}
+        - buttons: |
+            App|${BITRISE_APP_URL}
+            #mobile-testeng|https://mozilla.slack.com/archives/C02KDDS9QM9
+            Mana|https://mana.mozilla.org/wiki/display/MTE/Mobile+Test+Engineering
```
[the remark this answers is not in the document]
Comment (AaronMT) [reply bitrise.yml:1449]: It's not easily available, but we would have to wrap the call as a variable and then re-assign it with grep and use the above envman. I'll test it locally.

[the remark this answers is not in the document]
Comment (isabelrios) [reply bitrise.yml:1449]: oh I thought that having it in a similar way as we have it on [android projects](https://github.com/mozilla-mobile/firefox-android/blob/0a59a95289d618b622b574afb08bfa6afda1b181/fenix/automation/taskcluster/androidTest/parse-ui-test.py#L4) would be feasible.. if it is not and it takes a lot of effort / time, we can do that in follow up PR.

**Review by isabelrios — APPROVED** [review]
LGTM! looking forward to seeing this notifications!

File: bitrise.yml [hunk 3]
```diff
@@ -1419,14 +1419,38 @@ workflows:
...
-            --app "$BITRISE_IPA_PATH" \
-            --device=model=iphone13pro,version=15.7 \
-            --results-bucket=firefox_ios_test_artifacts \
-            --no-record-video \
-            --client-details=matrixLabel="Bitrise"
+            $HOME/google-cloud-sdk/bin/gcloud config set project "$GOOGLE_CLOUD_PROJECT"
+            GCLOUD_OUTPUT=$($HOME/google-cloud-sdk/bin/gcloud beta firebase test ios run \
+                --type robo \
+                --app "$BITRISE_IPA_PATH" \
+                --device=model=iphone13pro,version=15.7 \
```
Comment (isabelrios) [inline bitrise.yml:1426]: Wondering if we can add here another device

**Pull request merged**